core/src/main/java/com/linecorp/armeria/internal/PathAndQuery.java

```diff
@@ -471,7 +470,7 @@ private static void appendHexNibble(StringBuilder buf, int nibble) {
         int length;
 
         Bytes(int initialCapacity) {
-            data = new byte[initialCapacity];
+            data = ThreadLocalByteArray.get(initialCapacity);
```

Comment:
I'm assuming this usage is safe due to the way `Bytes` is used, but I feel this optimization is really scarry when the buffer is allowed to survive past a syntactic function block. But will leave it to you on whether to keep as is, refactor into one method somehow, or revert.

Reply:
Nevermind :)

Reply:
It was not safe because we're dealing with two `Bytes` instances, so I had to revert it. :sweat_smile: